File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.js
/**
 * @param {number[]} nums
 * @param {number[][]} queries
 * @return {number[]}
 */
const addEvenNums = function (arr){
    let sum = 0;
    for (let i = 0; i < arr.length; i++){
        if (arr[i] % 2 === 0) {
            sum += arr[i];
        }
    }
    return sum;
}

var sumEvenAfterQueries = function(nums, queries) {
    let result = [];
    for (let i = 0; i < queries.length; i++){
        nums[queries[i][1]] = nums[queries[i][1]] + queries[i][0]
        result.push(addEvenNums(nums))
    }
    return result;
};

File: 985-sum-of-even-numbers-after-queries/985-sum-of-even-numbers-after-queries.py
class Solution:
    def sumEvenAfterQueries(self, nums: List[int], queries: List[List[int]]) -> List[int]:
        evenSum = sum(x for x in nums if x % 2 == 0)
        result = []
        for val, i in queries:
            if nums[i] % 2 == 0:
                evenSum -= nums[i]
            nums[i] += val
            if nums[i] % 2 == 0:
                evenSum += nums[i]
            result.append(evenSum)
        return result